flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -310,8 +309,32 @@ private void flush() {
 
         inFlightRequestsCount++;
         submitRequestEntries(batch, requestResult);
-        numRecordsOutCounter.inc(batchSize);
+    }
+
+    /**
+     * Creates the next batch of request entries while respecting the {@code maxBatchSize} and
+     * {@code flushOnBufferSizeBytes}. Also adds these to the metrics counters.
```

Comment:
`flushOnBufferSizeBytes` incorrect variable name

Reply:
agh! my search didn't match this since it had no "In". Thanks for this one.